src/components/CIPipelineN/CIPipeline.tsx

```diff
@@ -529,7 +529,7 @@ export default function CIPipeline({
             _formDataErrorObj.name = validationRules.name(_formData.name)
             _formDataErrorObj[BuildStageVariable.Build].isValid = _formDataErrorObj.name.isValid
             let valid = _formData.materials.reduce((isValid, mat) => {
-                isValid = isValid && validationRules.sourceValue(mat.regex || mat.value).isValid
+                isValid = isValid && validationRules.sourceValue((mat.regex || mat.value), mat.type!=='WEBHOOK').isValid
```

Comment:
can't we simply say `mat.type !== 'WEBHOOK' || (isValid && validationRules.sourceValue(mat.regex || mat.value).isValid)`

Reply:
Instead of introducing shouldValidate like variable

Reply:
Nvm. Understood the idea behind this.